src/reporter.ts

```diff
@@ -45,38 +51,84 @@ export const shouldTrim = (spans, maxSendBytes: number): boolean => {
   );
 };
 
-export const forgeRequestBody = (spans, maxSendBytes): string | undefined => {
-  let resultSpans = [];
-
-  if (isPruneTraceOff() || !shouldTrim(spans, maxSendBytes)) {
-    return spans.length > 0 ? JSON.stringify(spans) : undefined;
-  }
-
-  logger.debug('Starting trim spans before send');
-
-  const functionEndSpan = spans[spans.length - 1];
-  const errorSpans = spans.filter((span) => spanHasErrors(span) && span !== functionEndSpan);
-  const normalSpans = spans.filter((span) => !spanHasErrors(span) && span !== functionEndSpan);
-
-  const orderedSpans = [...errorSpans, ...normalSpans];
-
-  let totalSize = getJSONBase64Size(resultSpans) + getJSONBase64Size(functionEndSpan);
+const isJsonContent = (payload: any, headers: Object) => {
+  return isString(payload) && headers['content-type'] && headers['content-type'].includes('json');
+};
 
-  for (let errorSpan of orderedSpans) {
-    let spanSize = getJSONBase64Size(errorSpan);
-    if (totalSize + spanSize < maxSendBytes) {
-      resultSpans.push(errorSpan);
-      totalSize += spanSize;
+function scrub(payload: any, headers: any, sizeLimit: number, truncated = false): string {
+  try {
+    if (isJsonContent(payload, headers)) {
+      if (truncated) payload = untruncateJson(payload);
+      return payloadStringify(JSON.parse(payload), sizeLimit);
     } else {
-      break;
+      return payloadStringify(payload, sizeLimit);
     }
+  } catch (e) {
+    return payloadStringify(payload, sizeLimit);
   }
-
-  resultSpans.push(functionEndSpan);
-
-  if (spans.length - resultSpans.length > 0) {
+}
+
+function scrubSpans(resultSpans: any[]) {
```

Comment:
Im not sure this is the right place for this function, this is should be only the reporter, scrubbing is part of the collection